truffleruby/src/main/c/cext/ruby.c

```diff
@@ -2291,23 +2291,34 @@ VALUE rb_get_path(VALUE object) {
 // Structs
 
 VALUE rb_struct_aref(VALUE s, VALUE idx) {
-  rb_tr_error("rb_struct_aref not implemented");
-  abort();
+  return (VALUE) truffle_invoke(RUBY_CEXT, "rb_struct_aref", s, idx);
 }
 
 VALUE rb_struct_aset(VALUE s, VALUE idx, VALUE val) {
-  rb_tr_error("rb_struct_aset not implemented");
-  abort();
+  return (VALUE) truffle_invoke(RUBY_CEXT, "rb_struct_aset", s, idx, val);
 }
 
 VALUE rb_struct_define(const char *name, ...) {
-  rb_tr_error("rb_struct_define not implemented");
-  abort();
+  VALUE *rb_name = name == NULL ? truffle_read(RUBY_CEXT, "Qnil") : rb_str_new_cstr(name);
```

Comment:
`truffle_read(RUBY_CEXT, "Qnil")` could just be `Qnil`. Don't worry about the extra function call overhead - we aren't doing performance yet, and anyway we have dynamic inlining!